### DISCUSSION TITLE
Is there a way to get Lazygit to use the hosting neovim for the commit editor?

### DISCUSSION THREAD

**Discussion by tristan957** [folke/snacks.nvim#360] [General]
It seems to open an embedded Neovim terminal and launch Neovim, but Neovim is available already since that is what Lazygit is running in. When editing a file, this situation works perfectly, but the commit editor does not, unfortunately.

**Comment by tristan957** [comment]
I tried playing with GIT_EDITOR in the snacks lazygit.env config, but it must either not work or I don't know the right incantation.

**Reply by xian-wen** [reply]
Have you set the `EDITOR=nvim` in your shell config file (.bashrc, .zshrc, .config/fish/config.fish etc.)? If yes, I think the commit editor will be Neovim. Below is mine:
![image](https://github.com/user-attachments/assets/464c2618-27c0-40e9-987a-55d80a8c95f6)

**Reply by tristan957** [reply]
Yes but you're running neovim in neovim which is what I'm referring to. It would be better to use the same neovim which launched lazygit

**Reply by xian-wen** [reply]
Duplicate with https://github.com/folke/snacks.nvim/issues/304. As Folke said, this is out of scope.

**Reply by tristan957** [reply]
Should we bring this up with lazygit upstream?

**Reply by xian-wen** [reply]
Maybe. Maybe this is also related to the `nvim-remote` preset?
